crates/holochain_integrity_types/src/info.rs

```diff
@@ -62,102 +62,138 @@ pub struct DnaInfo {
+                    })
+            })
+    }
+
+    /// Find the user type in the given iterator that matches this [`ScopedZomeType`].
+    pub fn find<I, K>(&self, iter: I, scoped_type: ScopedZomeType<T>) -> Option<I::Item>
+    where
+        I: IntoIterator<Item = K>,
+        K: Into<ZomeTypesKey<T>> + Copy,
+        T: PartialEq,
+    {
+        iter.into_iter()
+            .find_map(|key| (self.get(key)? == scoped_type).then(|| key))
+    }
+
+    /// Find the [`ZomeTypesKey`] for this [`ScopedZomeType`].
+    pub fn find_key(&self, scoped_type: ScopedZomeType<T>) -> Option<ZomeTypesKey<T>>
+    where
+        T: PartialEq,
+        T: From<u8>,
+    {
+        self.0
+            .iter()
+            .position(|(zome_id, _)| *zome_id == scoped_type.zome_id)
+            .and_then(|zome_index| {
+                // Safe to index because we just checked position.
+                self.0[zome_index]
+                    .1
+                    .iter()
+                    .position(|zome_type| *zome_type == scoped_type.zome_type)
+                    .and_then(|type_index| {
+                        Some(ZomeTypesKey {
+                            zome_index: u8::try_from(zome_index).ok()?.into(),
+                            type_index: u8::try_from(type_index).ok()?.into(),
+                        })
+                    })
+            })
     }
 
-    /// Convert a [`LocalZomeTypeId`] to a [`GlobalZomeTypeId`].
-    /// If the the [`LocalZomeTypeId`] maps to a [`GlobalZomeTypeId`] in scope for the calling zome,
-    /// then this will return a [`GlobalZomeTypeId`].
-    pub fn to_global_scope(&self, index: impl Into<LocalZomeTypeId>) -> Option<GlobalZomeTypeId> {
-        let index: LocalZomeTypeId = index.into();
-
-        let mut total_len: u8 = 0;
-
-        self.0.iter().find_map(|range| {
-            // Add the length of this range to the total length.
-            total_len = total_len.checked_add(range.end.0.checked_sub(range.start.0)?)?;
-
-            // If the total length is less than the local index, then we know
-            // this index is within our local scope.
-            (index.0 < total_len)
-                // Then calculate the offset from local to global scope.
-                .then(|| (range.end.0 as i16) - (total_len as i16))
-                // Then add the offset to the local index to get the global index.
-                .and_then(|offset| {
-                    let i = (index.0 as i16) + offset;
-                    Some(GlobalZomeTypeId(u8::try_from(i).ok()?))
-                })
-        })
+    /// Get all the [`ZomeId`] dependencies for the calling zome.
+    pub fn dependencies(&self) -> impl Iterator<Item = ZomeId> + '_ {
+        self.0.iter().map(|(zome_id, _)| *zome_id)
     }
 }
 
-impl From<u8> for GlobalZomeTypeId {
-    fn from(v: u8) -> Self {
-        Self(v)
+impl From<EntryDefIndex> for ZomeEntryTypesKey {
+    fn from(type_index: EntryDefIndex) -> Self {
```

Comment:
here's an example of why it seems like EntryDefIndex maybe just wants to be named EntryType